reflector.go

```diff
@@ -114,6 +114,7 @@ func (r *Reflector) Reflect(mappings []Mapping) error {
 				},
 			},
 			Data: k8sSecretData,
+			Type: mapping.SecretType,
```

Comment:
If you try to change the type of an existing secret, kubernetes will give you an error message. Something like:

`The Secret "foo" is invalid: type: Invalid value: "kubernetes.io/dockercfg": field is immutable`

So the procedure there would be to delete and recreate it. Is that how pentagon should handle it?

Reply:
Interesting point.

In general, I don't think I'd consider it safe to delete and re-create an actively-used secret, so transitions from one type to another should most likely be handled by having pentagon create a new secret (with a different name), then transitioning any uses over to the new one.

Reply:
I had this issue when testing this new feature as well.
A way could be to use the re-creation one only for this casuistic of existing secrets with a type miss-match. 
On the other hand, checking the jobs logs or setting a good monitoring which flag this issue should be a must at any production environment. 
All this being said, maybe adding a comment by now in the documentation should be enough if you don't want to get into this particular case tho.